🩹 fix(ci): cleanup-test-deployments workflow fails due to stderr capture in JSON mode

## Problem

The cleanup-test-deployments workflow (introduced in #316 to fix #209) has been failing consistently since deployment. All runs since Dec 1st fail with the same error:

```
jq: parse error: Invalid numeric literal at line 1, column 4
##[error]Process completed with exit code 5.
```

**Recent failure example:** https://github.com/codekiln/langstar/actions/runs/19865108855/job/56925547490

## Root Cause

The workflow script captures both stdout and stderr when calling `langstar graph list`:

```bash
if ! raw_deployments=$(./target/release/langstar graph list \
  --name-contains "test-deployment-" \
  --format json 2>&1); then
```

The `2>&1` redirection was added for error handling, but it has an unintended side effect. The `langstar` CLI correctly outputs informational messages to stderr when in JSON mode (as per cli/src/output.rs:104-105):

```rust
if self.format == OutputFormat::Json {
    eprintln!("{}", formatted);  // stderr for info messages
}
```

This means the captured output becomes:

```
ℹ Fetching deployments (limit: 20, offset: 0)...
{"resources":[...],"offset":0}
```

When this is piped to `jq`, it fails because the first line is not valid JSON, causing the "Invalid numeric literal at line 1, column 4" error.

## Evidence

1. **All recent runs fail with same error:**
   ```bash
   gh run list --workflow=cleanup-test-deployments.yml --limit 10
   # All 10 most recent runs show "failure" conclusion
   ```

2. **Workflow introduced:** PR #316 (merged Nov 26, 2025)
3. **First failures:** Started around Dec 1st, 2025
4. **Consistent error:** `jq: parse error: Invalid numeric literal at line 1, column 4`

## Solution

Remove the `2>&1` redirection for the `langstar graph list` command. In JSON mode, stderr is used for informational messages (by design), so we should not redirect it to stdout.

**Change from:**
```bash
if ! raw_deployments=$(./target/release/langstar graph list \
  --name-contains "test-deployment-" \
  --format json 2>&1); then
```

**To:**
```bash
if ! raw_deployments=$(./target/release/langstar graph list \
  --name-contains "test-deployment-" \
  --format json); then
```

This way:
- Actual errors still cause the command to fail (exit code)
- Info messages go to stderr (visible in workflow logs)
- Only valid JSON goes to stdout (captured in the variable)

## Alternative Considered

We could suppress info messages in JSON mode entirely, but the current design (info to stderr, JSON to stdout) follows Unix/CLI conventions used by cargo, git, curl, etc., and is the correct approach.

## Files Affected

- `.github/workflows/cleanup-test-deployments.yml` (line 56)

## Testing Plan

After fixing:
1. Manually trigger the workflow: `gh workflow run cleanup-test-deployments.yml`
2. Verify it completes successfully
3. Check that info messages still appear in logs (stderr)
4. Verify JSON parsing works correctly

## Related

- Original issue: #209
- Implementation PR: #316
- Workflow file: `.github/workflows/cleanup-test-deployments.yml`

--- a/.github/workflows/cleanup-test-deployments.yml
+++ b/.github/workflows/cleanup-test-deployments.yml
@@ -55,7 +55,7 @@ jobs:
           # Get test deployments as JSON with error handling, excluding cli-test-deployment-* deployments
           if ! raw_deployments=$(./target/release/langstar graph list \
             --name-contains "test-deployment-" \
-            --format json 2>&1); then
+            --format json); then
             echo "❌ Error: Failed to list deployments"
             echo "$raw_deployments"
             exit 1

